prov/util/src/util_mr_cache.c

```diff
@@ -38,6 +38,17 @@
 #include <ofi_mr.h>
 #include <ofi_list.h>
 
+static int util_mr_find_separate(void *a, void *b)
+{
+	struct iovec *iov1 = a, *iov2 = b;
+
+	if (ofi_iov_overlap_left(iov1, iov2))
+		return -1;
+	else if (ofi_iov_overlap_right(iov1, iov2))
+		return 1;
+	else
+		return 0;
+}
```

Comment:
> If the intent is not to merge regions, why not just search based on the iov_base address?

Sorry, I don't understand your idea here. Yes, the intent is not to merge regions, but we still need to find the regions that are within other regions. We need to know iov_len though.

> Why is a new function needed?

The function is needed to find region that is fully matched with searched region or searched region within a certain region.
The `util_mr_find_overlap` function searches overlapped regions that's should be ignored if we're not going to merge regions.

Reply:
The `util_mr_find_separate` function identifies the following (painted in blue):

![image](https://user-images.githubusercontent.com/11650339/38139715-536375ea-3439-11e8-819c-16f6859c3b13.png)

The `util_mr_find_overlap` function identifies the following (painted in blue)

![image](https://user-images.githubusercontent.com/11650339/38139869-f53ec2c0-3439-11e8-8502-8f50ea58c8af.png)


Agenda:
- green - search region
- blue - matched regions
- orange - not matched regions

Reply:
I think we need to rename this function too, since we're not looking for separate regions, but one that the current one can fit within.  Maybe util_mr_find_container() ?

Reply:
Or maybe util_mr_find_within() to match with ofi_iov_within()?

Reply:
> util_mr_find_within()

Yes, this fully reflects the idea of this function